Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -765,6 +761,8 @@ public bool ClearStateCache(QualifiedModuleName module, bool notifyStateChanged
                 {
                     keys.Add(key);
                 }
+
+                _errors.RemoveAll(w => w.ModuleName == key);
```

Comment:
That seems to be the wrong spot. This will remove all errors on modules presently known every time the state cache of some module gets cleared. 

This might be better off in `RemoveKeysFromCollections`.